### PR TITLE
Reservations controller methods

### LINKED ISSUE #10: [3pts] Create My-Reservation Api
- a GET request to get all reservations for a logged-in user
- a DELETE request to Delete a reservation that is not yet approved for a logged-in user

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR I implemented the below:

- Added my_reservation method to get all reservations made by a specific user
- Added delete_reservation method to delete a reservation as long as it's status is pending

closes https://github.com/ichala/short-let-api/issues/10

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -35,6 +35,28 @@ def handle_pending_reservation_admin
...
+
+  # delete reservation
+  def delete_reservation
+    if logged_in?
+      @reservation = current_user.reservations.find(params[:reservation_id])
+      raise ActiveRecord::RecordNotFound unless @reservation
+
+      destroy_reservation(@reservation)
+    else
+      render json: { error: 'You need to loginto access this page' }, status: :unauthorized
```
Comment (aimalamiri) [inline app/controllers/reservations_controller.rb]: Make sure to fix the spelling mistake in this message. `loginto`

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -35,6 +35,28 @@ def handle_pending_reservation_admin
     end
   end
 
+  # create my reservations for a user
+  def my_reservations
+    return render json: { error: 'You need to loginto access this page' }, status: :unauthorized unless logged_in?
+
+    @reservations = Reservation.find(user_id: 25)
```
Comment (ichala) [inline app/controllers/reservations_controller.rb]: i think you forget the user_id testing in here 😞 it should be
 ```ruby
@reservations = Reservation.find(user_id: current_user.id)
```

File: config/routes.rb [hunk 3]
```diff
@@ -8,6 +8,12 @@
   get "/authorized", to: "sessions#show"
   post "/reservation", to: "reservations#create"
 
+  #User Routes
+  #my reservations Routes
+  get "/user/reservations", to: "reservations#my_reservations"
+  #delete reservation
+  delete "/user/reservations", to: "reservations#delete_reservation"
```
Comment (ichala) [inline config/routes.rb]: [OPTIONAL] please add the correct indentation to follow the admin routes and make the routes readable and clean

File: app/controllers/reservations_controller.rb [hunk 4]
```diff
@@ -76,6 +98,18 @@ def create_reservation
...
+    if @reservation.status == 'Pending'
+      if reservation.destroy
+        render json: { message: 'reservation deleted' }, status: :ok
+      else
+        render json: { error: 'Reservation can\'t be destroyed' }, status: :not_implemented
+      end
+    else
+      render json: { error: 'You can not delete this reservation' }, status: :unauthorized
+    end
+  end
```
Comment (aimalamiri) [inline app/controllers/reservations_controller.rb]: [Optional] I like the idea of exporting code into multiple methods to make them reusable, but this method has no use case except the one in the above, so don't you think it's better to merge them back?

**Review by aimalamiri — CHANGES_REQUESTED** [review]
Hi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the this thread if something is not 100% clear.

**Review by ichala — CHANGES_REQUESTED** [review]
Hi @EidHachem , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No linters errors✅ 
- Clean Code✅ 

 ## Required Changes ♻️

- [x] _**check the comments below**_


## Optional suggestions

- - [x] _**check the comments below**_


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: app/controllers/reservations_controller.rb (hunk 4 above)
Comment (EidHachem) [reply app/controllers/reservations_controller.rb]: @aimalamiri  Done

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (EidHachem) [reply app/controllers/reservations_controller.rb]: @aimalamiri  Done

File: app/controllers/reservations_controller.rb (hunk 2 above)
Comment (EidHachem) [reply app/controllers/reservations_controller.rb]: @ichala Done with `Reservation.find(current_user.id)`

File: config/routes.rb (hunk 3 above)
Comment (EidHachem) [reply config/routes.rb]: @ichala Done

**Review by d2ndjim — APPROVED** [review]
Hello, @EidHachem 

Nice implementing the changes requested by @aimalamiri  and @ichala !
approve

* No linter error.
* Well structured code.
* Meeting all requirements

PR Review result:  APPROVED! 👍 👍 👍

**Review by aimalamiri — APPROVED** [review]
### **Hi 👋** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by ichala — APPROVED** [review]
Approved ✔️ congratz

**Pull request merged**